src/content_mapblock.cpp

```diff
@@ -925,6 +950,16 @@ void MapblockMeshGenerator::drawPlantlike()
 	draw_style = PLANT_STYLE_CROSS;
 	scale = BS / 2 * f->visual_scale;
 	offset = v3f(0, 0, 0);
+
+	NodeMetadata *meta = data->m_client->getEnv().getMap().getNodeMetadata(blockpos_nodes + p);
```

Comment:
this code is duplicated many time. It's a good candidate for a function which generate alter the offset